demo/c/orca_demo_streaming.c

```diff
@@ -0,0 +1,623 @@
+/*
+Copyright 2024 Picovoice Inc.
+
+You may not use this file except in compliance with the license. A copy of
+the license is located in the "LICENSE" file accompanying this source.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+License for the specific language governing permissions and limitations under
+the License.
+*/
+
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+
+#if !(defined(_WIN32) || defined(_WIN64))
+
+#include <dlfcn.h>
+
+#else
+
+#include <windows.h>
+
+#define UTF8_COMPOSITION_FLAG (0)
+#define NULL_TERMINATED       (-1)
+
+#endif
+
+#define DR_WAV_IMPLEMENTATION
+
+#include "dr_wav.h"
+
+#include "pv_orca.h"
+
+#define MAX_NUM_CHUNKS              (500)
+#define MAX_NUM_BYTES_PER_CHARACTER (5)
+
+static void *open_dl(const char *dl_path) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+    return LoadLibrary(dl_path);
+
+#else
+
+    return dlopen(dl_path, RTLD_NOW);
+
+#endif
+}
+
+static void *load_symbol(void *handle, const char *symbol) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+    return GetProcAddress((HMODULE) handle, symbol);
+
+#else
+
+    return dlsym(handle, symbol);
+
+#endif
+}
+
+static void close_dl(void *handle) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+    FreeLibrary((HMODULE) handle);
+
+#else
+
+    dlclose(handle);
+
+#endif
+}
+
+static void print_dl_error(const char *message) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+    fprintf(stderr, "%s with code `%lu`.\n", message, GetLastError());
+
+#else
+
+    fprintf(stderr, "%s with `%s`.\n", message, dlerror());
+
+#endif
+}
+
+static struct option long_options[] = {
+        {"access_key",   required_argument, NULL, 'a'},
+        {"library_path", required_argument, NULL, 'l'},
+        {"model_path",   required_argument, NULL, 'm'},
+        {"text",         required_argument, NULL, 't'},
+        {"output_path",  required_argument, NULL, 'o'},
+};
+
+static pv_status_t num_bytes_character(unsigned char c, int32_t *num_bytes) {
+    *num_bytes = 0;
+
+    int32_t nb;
+    if ((c & 0x80) == 0x00) {
+        nb = 1;
+    } else if ((c & 0xE0) == 0xC0) {
+        nb = 2;
+    } else if ((c & 0xF0) == 0xE0) {
+        nb = 3;
+    } else if ((c & 0xF8) == 0xF0) {
+        nb = 4;
+    } else {
+        return PV_STATUS_INVALID_ARGUMENT;
+    }
+
+    *num_bytes = nb;
+
+    return PV_STATUS_SUCCESS;
+}
+
+static double get_time() {
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    return (double) tv.tv_sec + ((double) tv.tv_usec * 1e-6);
+}
+
+static void print_usage(const char *program_name) {
+    fprintf(
+            stdout,
+            "Usage: %s [-l LIBRARY_PATH -m MODEL_PATH -a ACCESS_KEY -t TEXT -o OUTPUT_PATH]\n",
+            program_name);
+}
+
+typedef struct pcm_chunk pcm_chunk_t;
```

Comment:
the reason for having two-step definition is to keep the structure members private, which is not the case here so

```suggestion
typedef struct {
    int32_t num_samples;
    int16_t *pcm;
    pcm_chunk_t *next;
}  pcm_chunk_t;
```